microbench/src/test/java/io/netty/microbench/channel/ImmediateLocalServerChannel.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.microbench.channel;
+
+import io.netty.channel.ChannelProgressivePromise;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.local.LocalServerChannel;
+
+public class ImmediateLocalServerChannel extends LocalServerChannel {
+    @Override
+    public ChannelPromise newPromise() {
```

Comment:
Good point. it makes sense as PausableChannelEventLoop is of type EventExecutor.  not having to work around this would simplify things. I'll look into it but @trustin or @normanmaurer any objections on the surface?